api/client/webclient/webclient.go

```diff
@@ -67,10 +69,15 @@ func doWithFallback(clt *http.Client, allowPlainHTTP bool, req *http.Request) (*
 		return resp, nil
 	}
 
+	var proxy *url.URL
+	if proxy, err = http.ProxyFromEnvironment(req); err != nil {
```

Comment:
Ideally we wouldn't repeat the `http.ProxyFromEnvironment` function here and get it straight from the `http.Client`'s `Transport`. Unfortunately, `http.Transport` is an interface that doesn't allow us to get the proxy back out, as far as I can tell.